File: api/BeSureApi/Models/Receipt.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel;

namespace BeSureApi.Models
{
    public class ReceiptList
    {
            public int Id { get; set; }
            public string ReceiptNumber { get; set; }
            public decimal ReceiptAmount { get; set; }
         public string? CustomerName { get; set; }
        public string? PaymentMethod { get; set; }
        public DateTime? ReceiptDate { get; set; }
        
                  
    }

    public class ReceiptDetail
    {
        public int Id { get; set; }
        public string ReceiptNumber { get; set; }
        public decimal ReceiptAmount { get; set; }

    }

    public class InvoiceReceiptDetailCreate
    {
        public int InvoiceId { get; set; }
        public decimal Amount { get; set; }
    }

    public class ReceiptViewDetail
    {
        public int id { get; set; }
        public string ReceiptNumber { get; set; }
        public DateTime? ReceiptDate { get; set; }
        public string? CustomerName { get; set; }
        public string PaymentMethod { get; set; }
        public decimal? CollectedAmount { get; set; }
        public string? TransactionReferenceNumber { get; set; }
        public DateTime? ChequeDate { get; set; }
        public decimal? ChequeAmount { get; set; }
        public string? ChequePayeeName { get; set; }
        public string? ChequeBank { get; set; }
        public string? ChequeBranch { get; set; }
        public DateTime? ChequeReturnedOn { get; set; }
        public string? ChequeReturnedReason { get; set; }
        public DateTime? ChequeReceivedOn { get; set; }
        public string? TenantBankAccount { get; set; }
        public DateTime? ReceiptAmountRealisedOn { get; set; }
        public decimal ReceiptAmount { get; set; }
    }

    public class InvoiceReceiptList
    {
        public int Id { get; set; }
        public int InvoiceId { get; set; }
        public string InvoiceNumber { get; set; }
        public decimal InvoiceAmount { get; set; }
        public decimal ReceiptAmount { get; set; }
    }

    public class ReceiptViewWithDetail
    {
        public ReceiptViewDetail Receipt { get; set; }
        public List<InvoiceReceiptList>? InvoiceReceiptList { get; set; }
    }
}

File: ui/src/types/receipt.ts
import { Decoder, array, boolean, nullable, number, object, string } from "decoders";


export interface ReceiptListDetail {
  Id: number;
  ReceiptNumber: string;
  ReceiptAmount: number;
  CustomerName: string | null;
  PaymentMethod: string | null;
  ReceiptDate: string;
}

export const receiptListDetailDecoder: Decoder<ReceiptListDetail> = object({
  Id: number,
  ReceiptNumber: string,
  ReceiptAmount: number,
  CustomerName: nullable(string),
  PaymentMethod: nullable(string),
  ReceiptDate: string
});

export interface ReceiptList {
  Receipts: ReceiptListDetail[];
  TotalRows: number;
  PerPage:number;
}

export const receiptListDecoder: Decoder<ReceiptList> = object({
  Receipts: array(receiptListDetailDecoder),
  TotalRows: number,
  PerPage:number
});

export interface ReceiptView {
  ReceiptNumber: string | null;
  ReceiptDate: string | null;
  CustomerName: string | null;
  PaymentMethod: string | null;
  TransactionReferenceNumber: string | null;
  TenantBankAccount: string | null;
  ReceiptAmount: number | null;
}
export const receiptViewDecoder: Decoder<ReceiptView> = object({
  ReceiptNumber: nullable(string),
  ReceiptDate: nullable(string),
  CustomerName: nullable(string),
  PaymentMethod: nullable(string),
  TransactionReferenceNumber: nullable(string),
  TenantBankAccount: nullable(string),
  ReceiptAmount: nullable(number)
})


export interface InvoiceReceiptDetail {
  Id: number;
  InvoiceId: number;
  ReceiptAmount: number;
  InvoiceNumber: string;
  InvoiceAmount: number;
}

export const invoiceReceiptDetailDecoder: Decoder<InvoiceReceiptDetail> = object({
  Id: number,
  InvoiceId: number,
  ReceiptAmount: number,
  InvoiceNumber: string,
  InvoiceAmount: number
});
export interface ReceiptViewDetails {
  Receipt: ReceiptView;
  InvoiceReceiptList: InvoiceReceiptDetail[]
}

export const receiptViewDetailsDecoder: Decoder<ReceiptViewDetails> = object({
  Receipt: receiptViewDecoder,
  InvoiceReceiptList: array(invoiceReceiptDetailDecoder)
});

export interface InvoiceReceiptDetailCreate {
  InvoiceId: number;
  InvoiceNumber: string;
  Amount: number;
  PaidAmount: number;
  CollectionDueDate: string;
  PendingAmount: number;
}

export interface InvoiceReceiptListCreate {
  ContractInvoices: InvoiceReceiptDetailCreate[];
}